src/settings/panel.tsx

```diff
@@ -137,15 +136,10 @@ export class AiSettings extends React.Component<
     if (!this._secretsManager || !this._useSecretsManager) {
       return;
     }
-    // Attach the password inputs to the secrets manager only if they have changed.
-    const inputs = this._formRef.current?.getElementsByTagName('input') || [];
-    if (ArrayExt.shallowEqual(inputs, this._formInputs)) {
-      return;
-    }
```

Comment:
This comparison has been removed because, some time, when this function is called, some inputs are not yet in the DOM. This leads to a wrong comparison because the number of inputs did not change.
This shouldn't happen, according to the documentation of React. Adding a little timeout fixes it, so I guess there is a race condition (or something related to *RJSF*), but I couldn't fix it properly.

It occurs when updating the form after setting *HideSecretFields* value, which displays again the secrets fields.

It seems that the `await` below is enough for all the element to be in the DOM, but I wonder how robust it is.